src/edge/edgeManager.ts

```diff
@@ -383,12 +383,13 @@ export class EdgeManager {
         const modules = templateJson.modulesContent.$edgeAgent["properties.desired"].modules;
         let moduleName: string = "";
         let moduleInfo: ModuleInfo;
+        const marketplace = Marketplace.getInstance(this.context, Object.keys(modules));
         if (template === Constants.MARKETPLACE_MODULE) {
-            const marketplace = new Marketplace(this.context, Object.keys(modules));
             moduleInfo = await marketplace.importModule();
         } else {
             moduleName = Utility.getValidModuleName(await this.inputModuleName(targetModulePath, Object.keys(modules)));
             moduleInfo = await this.inputImage(moduleName, template);
+            marketplace.dispose();

```

Comment:
why dispose here?

Reply:
Consider one edge case: User open Marketplace page but does not do anything, then he import a C# module. The template json is changed. If the Marketplace page is not closed, he could still import the a Marketplace module, but the template/modules json info has already been updated with pervious C# module.